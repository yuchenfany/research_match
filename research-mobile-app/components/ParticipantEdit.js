/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */

import React, { useState, useEffect } from 'react';
import { View, Button, Text, StyleSheet } from 'react-native';
// import NavBar from './NavBar';

function ParticipantHome({ route, navigation }) { // add props user
  let {user, setUser} = route.params;
  console.log('PARTICIPANT EDIT === = ====');
  console.log(user.phys);
  console.log(user.password);
  
  const[tempUser, setTempUser] = useState({
      username: user.username,
      password: user.password, // NOTE: unhashes password
      age: user.age,
      // enrolled: user.enrolled,
      // age: user.age,
      // heightFeet: user.heightFeet,
      // heightInches: user.heightInches,
      // weight: user.weight,
      // sex: user.sex,
      // gender: user.gender,
      // allergies: user.allergies,
      // phys: user.phys,
      // psych: user.psych,
      // med: user.med,
      // type: user.type,
    }
  );

  const [ageErr, setAgeErr] = useState({ message: '' });
//   const [feetErr, setFeetErr] = useState({ message: '' });
//   const [inchErr, setInchErr] = useState({ message: '' });
//   const [weightErr, setWeightErr] = useState({ message: '' });

  const updateAge = async (event) => {
    console.log('EVENT: AGE');
    console.log(event.target.value);

    await setTempUser({
      username: tempUser.username,
      password: tempUser.password,
      age: event.target.value,
      // enrolled: tempUser.enrolled,
      // heightFeet: tempUser.heightFeet,
      // heightInches: tempUser.heightInches,
      // weight: tempUser.weight,
      // sex: tempUser.sex,
      // gender: tempUser.gender,
      // allergies: tempUser.allergies,
      // phys: tempUser.phys,
      // psych: tempUser.psych,
      // med: tempUser.med,
      // type: tempUser.type,
    });

    console.log('EVENT: AGE 2');
    console.log(tempUser.age);
    console.log(tempUser.username);
    console.log(tempUser.password);
    console.log(tempUser.sex);
  };

  const isValidInput = (input) => !(input.length === 0 || !input.match(/^[0-9]+$/));

  async function handleUpdate(event) {
    console.log('=============HANDLE UPDATE REACHED=============');

    // if (!isValidInput(user.age) || !isValidInput(user.weight) || !isValidInput(user.heightFeet)
    // || !isValidInput(user.heightInches)) {
    //   if (!isValidInput(user.age)) {
    //     setAgeErr({ message: 'Age: Enter a number' });
    //   } else {
    //     setAgeErr({ message: '' });
    //   }

    // //   if (!isValidInput(user.weight)) {
    // //     setWeightErr({ message: 'Weight: Enter a number' });
    // //   } else {
    // //     setWeightErr({ message: '' });
    // //   }

    // //   if (!isValidInput(user.heightFeet)) {
    // //     setFeetErr({ message: 'Feet: Enter a number' });
    // //   } else {
    // //     setFeetErr({ message: '' });
    // //   }

    // //   if (!isValidInput(user.heightInches)) {
    // //     setInchErr({ message: 'Inches: Enter a number' });
    // //   } else {
    // //     setInchErr({ message: '' });
    // //   }

    //   event.preventDefault();

    //   return;
    // }


    if (await postUserInfo()) {
        await setUser(tempUser);

        navigation.navigate('ParticipantHome', {
            user,
            setUser,
          });    
    } else {
      event.preventDefault();
    }
  }

  async function postUserInfo() {
    console.log('TEMP USER LOG');
    // console.log(JSON.stringify(tempUser));
    await fetch(`http://localhost:5000/record/participant-edit/${tempUser.username}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(tempUser),
    })
      .catch((e) => {
        window.alert(e);
      });

    return true;
  }

//   useEffect(() => {
//     getAllStudyJson()
//       .then(setEnrolledStudies);
//   }, []);

//   function goToStudy(studyId) {
//     console.log(studyId);
//     setStudy({ studyId });
//     navigate(`/study/${studyId}`);
//   }

  return (
    <View>
        <Text>EDIT PROFILE</Text>
        <View>
          <div>Age</div>
          <input
            className="small-input"
            type="text"
            id="age"
            value={user.age}
            onChange={updateAge}
          />
          <span className="error-message">{ageErr.message}</span>
        </View>
        <Button title="UPDATE" onPress={() => handleUpdate()}/>
    </View>
  );
}

export default ParticipantHome;
