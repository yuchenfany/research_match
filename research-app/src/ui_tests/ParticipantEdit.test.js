/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import ParticipantEdit from '../components/ParticipantEdit';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

test('Title Edit User Profile', () => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };  
    render(
        <Router>
            <ParticipantEdit user={user1} setUser={mockSetUser} />
        </Router>);  
  const label = screen.getByText("Edit User Profile");
  expect(label).toBeInTheDocument();
});

test('find allergies text', () => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };  
    render(
        <Router>
            <ParticipantEdit user={user1} setUser={mockSetUser} />
        </Router>);  
  const label = screen.getByText("Allergies");
  expect(label).toBeInTheDocument();
});

test('find Medications text', () => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };    
    render(
        <Router>
            <ParticipantEdit user={user1} setUser={mockSetUser} />
        </Router>);  
  const label = screen.getByText("Medications");
  expect(label).toBeInTheDocument();
});

test('Update Button',() => {
  const Button = ({onClick, children}) => (
    <button onClick={onClick}>{children}</button>
  )
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };    
      render(
        <Router>
            <ParticipantEdit user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /update/i,
  });
  fireEvent.click(screen.getByText(/update/i));
  expect(linkElement).toBeInTheDocument();
});

test('Cancel Button',() => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };
  render(
        <Router>
            <ParticipantEdit user={user1} setUser={mockSetUser} />
        </Router>);  
  const linkElement = screen.getByRole('button', {
    name: /cancel/i,
  });
  fireEvent.click(linkElement);
  expect(linkElement).toBeInTheDocument();
});

test('single input value  sequence', () => {
  const mockUser = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };
  const mockSetUser = jest.fn();

  render(
    <Router>
      <ParticipantEdit user={mockUser} setUser={mockSetUser} />
    </Router>,
  );

  const input = screen.getByPlaceholderText('age');
  fireEvent.change(input, { target: { value: '10' } });
  expect(input.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input2 = screen.getByPlaceholderText('height');
  fireEvent.change(input2, { target: { value: '10' } });
  expect(input2.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input3 = screen.getByPlaceholderText('ft');
  fireEvent.change(input2, { target: { value: '10' } });
  expect(input3.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  const input4 = screen.getByPlaceholderText('weight');
  fireEvent.change(input2, { target: { value: '10' } });
  expect(input4.value).toEqual('10');
  // expect(mockSetUser).toHaveBeenCalled();

  // fireEvent.click(screen.getByDisplayValue('SUBMIT'));
});

test('radio buttons', async () => {
  const mockUser = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };
  const mockSetUser = jest.fn();

  render(
    <Router>
      <ParticipantEdit user={mockUser} setUser={mockSetUser} />
    </Router>,
  );

  await userEvent.click(screen.getByTitle('intersex-form'));
});

// const genderOptions = [
//   { label: 'Female', value: 'female' },
//   { label: 'Male', value: 'male' },
//   { label: 'Transgender', value: 'transgender' },
//   { label: 'Non-binary', value: 'nonbinary' },
//   { label: 'Other', value: 'other' },
//   { label: 'Prefer not to answer', value: 'prefer not to answer' },
// ];
// /* eslint-disable */
// jest.mock('react-select', () => ({ options, value, onChange }) => {
//   function handleChange(event) {
//     const option = options.find(
//       (option) => option.value === event.currentTarget.value,
//     );
//     onChange(option);
//   }

//   function isGender(curr) {
//     if (curr[0].value === 'female') {
//       return "gender";
//     }
//     return "not-gender";
//   }

//   return (
//     <select data-testid={isGender(options)} value={value} onChange={handleChange}>
//       {options.map(({ label, value }) => (
//         <option key={value} value={value}>
//           {label}
//         </option>
//       ))}
//     </select>
//   );
// });
// /* eslint-enable */

// test('Edit Study tags', () => {
//   const mockSetUser = jest.fn();
//   const mockUser = {
//     username: 'user1',
//     password: 'user1pass',
//     enrolled: '',
//     age: '10',
//     heightFeet: '10',
//     heightInches: '10',
//     weight: '10',
//     sex: 'male',
//     gender: 'male',
//     allergies: '',
//     phys: '',
//     psych: [],
//     med: '',
//     type: 0,
//   };

//   render(
//     <Router>
//       <ParticipantEdit user={mockUser} setUser={mockSetUser} />
//     </Router>,
//   );
//   const select = screen.getByTestId('gender');
//   fireEvent.change(select, {
//     target: { value: 'Female' },
//   });
// });

test('snapshot test', () => {
  const mockSetUser = jest.fn();
  const user1 = {
    username: 'user1',
    password: 'user1pass',
    enrolled: '',
    age: '10',
    heightFeet: '10',
    heightInches: '10',
    weight: '10',
    sex: 'male',
    gender: 'male',
    allergies: '',
    phys: '',
    psych: [],
    med: '',
    type: 0,
  };

  const notification = false;
  const mockSetStudy = jest.fn();
  const study = {
    studyId: '',
    participants: [''],
    researchers: [],
    tags: [],
  };
  const component = renderer.create(
    <Router>
      <ParticipantEdit user={user1} setUser={mockSetUser} />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// TO GET HIGHER SCORE

// 1. fireevent update button is not accessing
// 2. text fields for name and organization
