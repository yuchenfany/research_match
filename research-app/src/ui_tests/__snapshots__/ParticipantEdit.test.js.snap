// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="ParticipantEdit"
>
  <div
    className="nav"
  >
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      HOME
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MY STUDIES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      MESSAGES
    </button>
    <button
      className="nav-btn"
      onClick={[Function]}
      type="button"
    >
      ANALYTICS
    </button>
    <button
      className="account-btn"
      onClick={[Function]}
      type="button"
    >
      ACCOUNT
    </button>
    <div
      className="acct-options"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        EDIT PROFILE
      </button>
      <button
        className="account-btn"
        onClick={[Function]}
        type="button"
      >
        DELETE ACCOUNT
      </button>
      <button
        className="logout-btn"
        onClick={[Function]}
        type="button"
      >
        LOGOUT
      </button>
    </div>
  </div>
  <div
    className="profile-flex"
  >
    <div
      className="header-left"
    >
       Edit User Profile 
    </div>
    <div
      className="profile-row"
    >
      <div>
        Age
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="age"
        type="text"
        value="10"
      />
      <span
        className="error-message"
      >
        
      </span>
      <div>
        Height
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="height"
        type="text"
        value="10"
      />
      <div>
        ft
      </div>
      <span
        className="error-message"
      >
        
      </span>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="ft"
        type="text"
        value="10"
      />
      <div>
        in
      </div>
      <span
        className="error-message"
      >
        
      </span>
      <div>
        Weight
      </div>
      <input
        className="small-input"
        id="age"
        onChange={[Function]}
        placeholder="weight"
        type="text"
        value="10"
      />
      <div>
        lbs
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Biological Sex
      </div>
      <label
        className="radio-option"
        htmlFor="form"
      >
        <input
          defaultChecked="checked"
          id="male"
          name="option"
          onClick={[Function]}
          type="radio"
          value="male"
        />
        <div>
          Male
        </div>
      </label>
      <label
        className="radio-option"
        htmlFor="form"
      >
        <input
          defaultChecked=""
          id="female"
          name="option"
          onClick={[Function]}
          type="radio"
          value="female"
        />
        <div>
          Female
        </div>
      </label>
      <label
        className="radio-option"
        htmlFor="form"
        title="intersex-form"
      >
        <input
          defaultChecked=""
          id="intersex"
          name="option"
          onClick={[Function]}
          type="radio"
          value="intersex"
        />
        <div>
          Intersex
        </div>
      </label>
    </div>
    <div
      className="profile-row"
    >
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Gender
      </div>
      <div
        className="dropdown"
      >
        <div
          className=" css-b62m3t-container"
          id="gender"
          onKeyDown={[Function]}
        >
          <span
            className="css-1f43avz-a11yText-A11yText"
            id="react-select-37-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            className="css-1f43avz-a11yText-A11yText"
          />
          <div
            className=" css-108aapi-control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className=" css-319lph-ValueContainer"
            >
              <div
                className=" css-qc6sy-singleValue"
              >
                Male
              </div>
              <div
                className=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-select-37-listbox"
                  aria-expanded={false}
                  aria-haspopup={true}
                  aria-owns="react-select-37-listbox"
                  autoCapitalize="none"
                  autoComplete="off"
                  autoCorrect="off"
                  className=""
                  disabled={false}
                  id="react-select-37-input"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  role="combobox"
                  spellCheck="false"
                  style={
                    Object {
                      "background": 0,
                      "border": 0,
                      "color": "inherit",
                      "font": "inherit",
                      "gridArea": "1 / 2",
                      "label": "input",
                      "margin": 0,
                      "minWidth": "2px",
                      "opacity": 1,
                      "outline": 0,
                      "padding": 0,
                      "width": "100%",
                    }
                  }
                  tabIndex={0}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                className=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                className=" css-tlfecz-indicatorContainer"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <svg
                  aria-hidden="true"
                  className="css-tj5bde-Svg"
                  focusable="false"
                  height={20}
                  viewBox="0 0 20 20"
                  width={20}
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Allergies
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className=" css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-38-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-38-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-38-listbox"
                aria-describedby="react-select-38-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-38-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-38-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medical Conditions (Physical)
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-39-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-39-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-39-listbox"
                aria-describedby="react-select-39-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-39-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-39-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medical Conditions (Psychological)
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-40-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-40-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-40-listbox"
                aria-describedby="react-select-40-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-40-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-40-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="profile-row"
    >
      <div>
        Medications
      </div>
      <span
        className="error-message"
      >
        
      </span>
    </div>
    <div
      className="profile-row"
    >
      <div
        className="select-tags css-b62m3t-container"
        onKeyDown={[Function]}
      >
        <span
          className="css-1f43avz-a11yText-A11yText"
          id="react-select-41-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          className="css-1f43avz-a11yText-A11yText"
        />
        <div
          className=" css-108aapi-control"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className=" css-319lph-ValueContainer"
          >
            <div
              className=" css-14el2xx-placeholder"
              id="react-select-41-placeholder"
            >
              Select...
            </div>
            <div
              className=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-41-listbox"
                aria-describedby="react-select-41-placeholder"
                aria-expanded={false}
                aria-haspopup={true}
                aria-owns="react-select-41-listbox"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                className=""
                disabled={false}
                id="react-select-41-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                role="combobox"
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "color": "inherit",
                    "font": "inherit",
                    "gridArea": "1 / 2",
                    "label": "input",
                    "margin": 0,
                    "minWidth": "2px",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "100%",
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              className=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              className=" css-tlfecz-indicatorContainer"
              onMouseDown={[Function]}
              onTouchEnd={[Function]}
            >
              <svg
                aria-hidden="true"
                className="css-tj5bde-Svg"
                focusable="false"
                height={20}
                viewBox="0 0 20 20"
                width={20}
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="button-row"
    >
      <input
        className="cancel-button"
        onClick={[Function]}
        type="submit"
        value="CANCEL"
      />
      <input
        className="update-button"
        onClick={[Function]}
        type="submit"
        value="UPDATE"
      />
    </div>
  </div>
</div>
`;
